src/machine/lib_machine.rs

```diff
@@ -524,4 +531,82 @@ mod tests {
             ),]))
         );
     }
+
+    #[test]
```

Comment:
I would say this PR (and the ping-pong between me and Mark) demonstrates that the tests here are, first and foremost, covering the function `run_query()` in this same file. So I would suggest to keep them here because the context of the test be located near `run_query()` provides import meaning, independent of the visibility of that function.

That said, I'm happy to move it over to `tests/` if that's where you want it to be :)
(just explaining my reasoning)

You are talking about all tests in this file?

Reply:
I have invited you to our forked repo so you can push changes to this branch if you like, @Skgland!

Reply:
I think I had some misconception/misunderstanding while writing that comment.

While I most of these would make sense as integration tests, as they appear to only use the public interface, I think most are small enough to be fine as is, especially as most already existed here from before this PR.

Only `integration_test` due to the now huge `lib_integration_test_commands.txt` (12k+ lines) appears too large.
Maybe the txt file could be split into multiple smaller txt files, so that the files can still be viewed here on Github?
Not sure how intertwined everything in there is.

`integration_test` could then be split into multiple test functions including the individual txt files which would then call a helper function that is basically the current `integration_test` but taking `code` as an argument.

Reply:
I think it's adequate to question if this test should be here at all. It was a pragmatic way for us to delineate where the problems are stemming from as we did some refactoring in our project at the same time we switched from SWI to Scryer, but had our integration tests from before.

Running this big txt file here is a bit opaque, which is also why I tried to extract the failing assertions into the other more understandable test cases. But it did already uncover a problem that you only see if you have multiple consecutive calls to `consult` followed by queries. This is actually close the reason that made us add this test mechanism: we saw scryer slowing down query by query when used in our ad4m integration tests (that's why we had the previous version without assertions on the results - it was just making sure Scryer would make it to the end). So I would argue against splitting it up.

But yeah, that does make it feel more like an integration test.
I'll leave it completely up to you guys to decide if you want to keep this test at all, and if so where it should stay.

Reply:
I think its not worth blocking the whole PR about this, as it can always be moved later.